### PR TITLE
feat/fix💥: adjust and fix various issues with scheduled events

### PR DESCRIPTION
**Pull request by AstreaTSS**
## Pull Request Type

- [ ] Feature addition
- [x] Bugfix
- [ ] Documentation update
- [ ] Code refactor
- [ ] Tests improvement
- [ ] CI/CD pipeline enhancement
- [ ] Other: [Replace with a description]

## Description
This PR improves using scheduled events in the library by fixing longstanding issues related to it and its events. In particular, this adds a proper method for fetching scheduled events in the cache, adds a get event for scheduled events for guilds, and fixes `GuildScheduledEventUserAdd/Remove` since those were utterly broken.

Note: this *technically* is breaking, and so should have some eyes on it. However, `GuildScheduledEventUserAdd/Remove` is so broken and unusable in its current state to the point where you would *have* to use a raw event to get any useful information most of the time. I understand that violates our versioning standards, but I legitimately think this is such an extreme case that it should be considered.

## Changes
- 💥 Fix `GuildScheduledEventUserAdd/Remove` by properly passing the IDs instead of the objects of the events.
- Fix the get scheduled event HTTP method to properly pass in `with_user_count`.
- Add in `fetch_scheduled_event` to `SmartCache`.
  - The `Guild` variant now uses the above method, using the cache properly.
- Add in `get_scheduled_event` to `Guild`.
- Properly typehint `user_count` in `ScheduledEvent`.

## Python Compatibility

- [ ] I've ensured my code works on Python `3.10.x`
- [x] I've ensured my code works on Python `3.11.x`

## Checklist

- [x] I've run the `pre-commit` code linter over all edited files
- [x] I've tested my changes on supported Python versions
- [ ] I've added tests for my code, if applicable
- [ ] I've updated / added  documentation, where applicable

### CODE REVIEW

**Review by i0bs — APPROVED** [review]

**Comment by i0bs** [comment]
I cannot access Discord, but I believe decision making of placing this on hold up to @LordOfPolls since it's an obvious breaking change.

**Comment by LordOfPolls** [comment]
Merging as non breaking - you cant break the api for something that never worked to start with

**Pull request merged**
